### PR TITLE
Add quote to command substitutions

### PR DESCRIPTION
**Pull request by haiyanmeng**
This is to follow
https://google.github.io/styleguide/shellguide.html#quoting

### CODE REVIEW

File: ar-setup.sh [hunk 1]
```diff
@@ -121,7 +121,7 @@ add_gitlab_project_vars() {
 	gitlab_project=${gitlab_project//\//%2F}
 
 	# check whether the GitLab project exists
-	output_project_existence=$(curl -s --header "PRIVATE-TOKEN: ${gitlab_access_token}" "${gitlab_hostname}api/v4/projects/${gitlab_project}")
+	output_project_existence="$(curl -s --header "PRIVATE-TOKEN: ${gitlab_access_token}" "${gitlab_hostname}api/v4/projects/${gitlab_project}")"
```
Comment (nan-yu) [inline ar-setup.sh:124]: nested quotes. You need to escape the nested ones.

File: ar-setup.sh [hunk 2]
```diff
@@ -131,23 +131,23 @@ add_gitlab_project_vars() {
 	fi
 
 	# add the AR repo and the service account key into the app project as GitLab project variables
-	output_repo=$(curl -s --header "PRIVATE-TOKEN: ${gitlab_access_token}" "${gitlab_hostname}api/v4/projects/${gitlab_project}/variables/${project_var_name_gcp_ar_repo}")
-	output_key=$(curl -s --header "PRIVATE-TOKEN: ${gitlab_access_token}" "${gitlab_hostname}api/v4/projects/${gitlab_project}/variables/${project_var_name_gcp_ar_key}")
+	output_repo="$(curl -s --header "PRIVATE-TOKEN: ${gitlab_access_token}" "${gitlab_hostname}api/v4/projects/${gitlab_project}/variables/${project_var_name_gcp_ar_repo}")"
+	output_key="$(curl -s --header "PRIVATE-TOKEN: ${gitlab_access_token}" "${gitlab_hostname}api/v4/projects/${gitlab_project}/variables/${project_var_name_gcp_ar_key}")"
```
Comment (nan-yu) [inline ar-setup.sh:135]: ditto for nested quotes

File: ar-setup.sh [hunk 3]
```diff
@@ -131,23 +131,23 @@ add_gitlab_project_vars() {
...
 		# For each service account, only 12 keys can be created. So we only create the key when all the preconditions are met.
-		key_file=$(mktemp)
+		key_file="$(mktemp)"
 		gcloud iam service-accounts keys create --iam-account="${service_account_email}" "${key_file}"
 		if [[ $? -ne 0 ]]; then
 			echo "Failed to download a service account key for ${service_account_email}"
 			exit 1
 		fi
-		service_account_key=$(cat "${key_file}")
+		service_account_key="$(cat "${key_file}")"
```
Comment (nan-yu) [inline ar-setup.sh:146]: you can remove the quotes around ${key_file}

File: ar-setup.sh [hunk 4]
```diff
@@ -131,23 +131,23 @@ add_gitlab_project_vars() {
...
 		rm -f "${key_file}"
 		artifact_repo_name="${artifact_registry_location}-docker.pkg.dev/${project}/${app_name}"
 
-		output=$(curl -s --request POST --header "PRIVATE-TOKEN: ${gitlab_access_token}" "${gitlab_hostname}api/v4/projects/${gitlab_project}/variables" --form "key=${project_var_name_gcp_ar_repo}" --form "value=${artifact_repo_name}")
+		output="$(curl -s --request POST --header "PRIVATE-TOKEN: ${gitlab_access_token}" "${gitlab_hostname}api/v4/projects/${gitlab_project}/variables" --form "key=${project_var_name_gcp_ar_repo}" --form "value=${artifact_repo_name}")"
```
Comment (nan-yu) [inline ar-setup.sh:150]: nested quotes

File: ar-setup.sh (hunk 1 above)
Comment (haiyanmeng) [reply ar-setup.sh:124]: Check https://google.github.io/styleguide/shellguide.html#quoting:
```
# "quote command substitutions"
# Note that quotes nested inside "$()" don't need escaping.
flag="$(some_command and its args "$@" 'quoted separately')"
```

Comment (nan-yu) [reply ar-setup.sh:124]: Oh, okay. Good to know no escaping is needed. Please ignore my other comments.

**Review by janetkuo — APPROVED** [review]

**Pull request merged**